### DISCUSSION TITLE
Maya 2022

### DISCUSSION THREAD

**Discussion by kreppene** [Autodesk/maya-usd#3659] [Q&A] [answered]
Hi everyone,
are all versions of mayaUsd for maya 2022 still on Asset Resolver v1 ?

In the pxr/usd/ar/resolver.h file it looks like it should be able to use the AR_VERSION 2, but the resolver_v2.h file is missing from the included dev kit

**Comment by seando-adsk** [comment]
Actually all our Maya 2022 and 2023 builds use our custom USD v21.11 build which is using AR1.
We only enabled AR2 in our USD v22.11 build which is used in Maya 2024.

Sean

**Reply by kreppene** [reply]
Is there a reason for keeping maya 2022/2023 on v1? 
Should it be possible to compile a custom build of mayaUsd with AR 2?

**Comment by wallworm** [comment, marked as answer]
To maintain a consistent USD ecosystem throughout our product portfolio and 3rd party vendors, we chose to lock down on a single USD version for that year. Our only exception to this is when we have to make new USD versions due to security patches.

When it comes to utilizing new features in older versions, you can choose to rebuild Maya USD with your own requirements. That is one of the beauties of being Open Source--you are not tied to what we are offering at this time.

That being said, we cannot guarantee how far you will get when making custom builds. We only officially support the versions that we release.

**Reply by kreppene** [reply]
Thanks, that makes sense.  

Maybe this is a good opportunity to upgradee to the latest maya anyway ;)

Cheers
